src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -231,6 +229,14 @@ public int getNumberOfKeyOccurrences(String key) {
         return duplicationChecker.getNumberOfKeyOccurrences(key);
     }
 
+    /**
+     * Sets the given key to the given entry.
+     * If the key is null, the entry field will be cleared.
+     *
+     * @param entry a BibEntry
+     * @param key a String
+     * @return true, if the entry contains the key, false if not
+     */
     public synchronized boolean setCiteKeyForEntry(BibEntry entry, String key) {
         String oldKey = entry.getCiteKeyOptional().orElse(null);
         if (key == null) {
```

Comment:
Why isn't `!Optional.isPresent()` used? The hack to convert to `null` is dirty. Optional has been introduced to avoid `null` at al..